controllers/watchers/watcher_test.go

```diff
@@ -0,0 +1,316 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/DataDog/chaos-controller/controllers/watchers"
+	"github.com/DataDog/chaos-controller/controllers/watchers/mocks"
+	logger "github.com/DataDog/chaos-controller/log"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8scache "sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var _ = Describe("watcher", func() {
+	const watcherName = "watcher-name"
+
+	var (
+		watcher          watchers.Watcher
+		cacheMock        mocks.CacheMock
+		informerMock     mocks.InformerMock
+		handlerMock      mocks.HandlerMock
+		targetObjectType client.Object
+		err              error
+		cacheContextFunc func() (ctx context.Context, cancel context.CancelFunc)
+	)
+
+	namespaceName := types.NamespacedName{Name: "namespace-name", Namespace: "namespace"}
+
+	BeforeEach(func() {
+		targetObjectType = &v1.Pod{}
+	})
+
+	JustBeforeEach(func() {
+		// Arrange
+		log, err := logger.NewZapLogger()
+		Expect(err).ShouldNot(HaveOccurred())
```

Comment:
Why not create a single logger in test setup, instead of one per test case?

Reply:
Of course! It has no sense to be repeated each time.